src/msg/msg_types.cc

```diff
@@ -129,9 +129,15 @@ bool entity_addr_t::parse(const char *s, const char **end, int default_type)
   if (inet_pton(AF_INET, buf4, &a4)) {
     u.sin.sin_addr.s_addr = a4.s_addr;
     u.sa.sa_family = AF_INET;
+#if defined(__FreeBSD__)
+    u.sa.sa_len = sizeof(u.sin);
+#endif
     p = start + strlen(buf4);
   } else if (inet_pton(AF_INET6, buf6, &a6)) {
     u.sa.sa_family = AF_INET6;
+#if defined(__FreeBSD__)
```

Comment:
instead of hardwiring this to `defined(__FreeBSD__)`. i'd suggest check the `sa_len` using

```cmake
check_struct_has_member("struct sockaddr" sa_len sys/socket.h HAVE_STRUCT_SOCKADDR_SA_LEN LANGUAGE C)
```

Reply:
@tchaikov 
Ah, new Cmake trick.... Will try to work that out.